rmm/src/event/mod.rs

```diff
@@ -69,8 +69,31 @@ impl Context {
         F: Fn(&[usize], &mut [usize]) -> Result<(), Error>,
     {
         self.ret[0] = rmi::SUCCESS;
-        if let Err(code) = handler(&self.arg[..], &mut self.ret[..]) {
-            self.ret[0] = code.into();
+
+        #[cfg(feature = "stat")]
+        {
+            if self.cmd != rmi::REC_ENTER {
+                trace!("let's get STATS.lock() with cmd {}", rmi::to_str(self.cmd));
+                let mut stats = crate::stat::STATS.lock();
+
+                if let Err(e) = stats.start(self.cmd) {
+                    info!("stats.start is failed: {:?} with cmd {:x}", e, self.cmd);
+                }
+                if let Err(code) = handler(&self.arg[..], &mut self.ret[..]) {
+                    self.ret[0] = code.into();
+                }
+                if let Err(e) = stats.end(self.cmd) {
+                    info!("stats.end is failed: {:?} with cmd {:x}", e, self.cmd);
+                }
```

Comment:
How about expose only `fn measure` in stat module like below? If then, we can abstract locking and others function.

```suggestion
stat::measure(self.cmd, || {
  if let Err(code) = handler(&self.arg[..], &mut self.ret[..]) {
    self.ret[0] = code.into();
   }
});
```

Reply:
That's a good idea! I'll try it. thank you :) !

Reply:
I reflected it :)